src/lib/certificate-manager.ts

```diff
@@ -0,0 +1,333 @@
+import { getUserDataCertificatesPath } from 'src/storage/paths';
+import { shellOpenExternalWrapper } from './shell-open-external-wrapper';
+
+// Certificate configuration
+const CA_NAME = 'WordPress Studio CA';
+const CA_CERT_VALIDITY_DAYS = 3650; // 10 years
+const SITE_CERT_VALIDITY_DAYS = 825; // a little over 2 years
+const CERT_DIRECTORY = getUserDataCertificatesPath();
+const CA_CERT_PATH = path.join( CERT_DIRECTORY, 'studio-ca.crt' );
+const CA_KEY_PATH = path.join( CERT_DIRECTORY, 'studio-ca.key' );
+
+// Make sure the certificates directory exists
+if ( ! fs.existsSync( path.join( CERT_DIRECTORY, 'domains' ) ) ) {
+	fs.mkdirSync( path.join( CERT_DIRECTORY, 'domains' ), { recursive: true } );
+}
+
+/**
+ * Generate a root CA certificate if it doesn't exist
+ */
+export async function ensureRootCA(): Promise< { cert: string; key: string } > {
+	// Check if CA cert and key already exist
+	if ( fs.existsSync( CA_CERT_PATH ) && fs.existsSync( CA_KEY_PATH ) ) {
+		return {
+			cert: fs.readFileSync( CA_CERT_PATH, 'utf8' ),
+			key: fs.readFileSync( CA_KEY_PATH, 'utf8' ),
+		};
+	}
+
+	// Generate a new CA certificate
+	console.log( 'Generating new root CA certificate...' );
+
+	// Generate a key pair
+	const keys = forge.pki.rsa.generateKeyPair( 2048 );
+
+	// Create a certificate
+	const cert = forge.pki.createCertificate();
+	cert.publicKey = keys.publicKey;
+	cert.serialNumber = '01';
+	cert.validity.notBefore = new Date();
+	cert.validity.notAfter = new Date();
+	cert.validity.notAfter.setDate( cert.validity.notBefore.getDate() + CA_CERT_VALIDITY_DAYS );
+
+	// Set certificate attributes
+	const attrs = [
+		{ name: 'commonName', value: CA_NAME },
+		{ name: 'countryName', value: 'US' },
+		{ name: 'organizationName', value: 'WordPress Studio' },
+		{ name: 'organizationalUnitName', value: 'Development' },
+	];
+	cert.setSubject( attrs );
+	cert.setIssuer( attrs );
+
+	// Set extensions
+	cert.setExtensions( [
+		{
+			name: 'basicConstraints',
+			cA: true,
+		},
+		{
+			name: 'keyUsage',
+			keyCertSign: true,
+			digitalSignature: true,
+			nonRepudiation: true,
+			keyEncipherment: true,
+			dataEncipherment: true,
+		},
+		{
+			name: 'extKeyUsage',
+			serverAuth: true,
+			clientAuth: true,
+		},
+		{
+			name: 'nsCertType',
+			client: true,
+			server: true,
+			email: true,
+			objsign: true,
+			sslCA: true,
+			emailCA: true,
+			objCA: true,
+		},
+	] );
+
+	// Self-sign the certificate
+	cert.sign( keys.privateKey, forge.md.sha256.create() );
+
+	// Convert to PEM format
+	const certPem = forge.pki.certificateToPem( cert );
+	const keyPem = forge.pki.privateKeyToPem( keys.privateKey );
+
+	// Save to files
+	fs.writeFileSync( CA_CERT_PATH, certPem );
+	fs.writeFileSync( CA_KEY_PATH, keyPem );
+
+	// Trust the CA certificate
+	await trustRootCA();
+
+	return { cert: certPem, key: keyPem };
+}
+
+export async function openCertificate() {
+	shellOpenExternalWrapper( `file://${ CA_CERT_PATH }` );
```

Comment:
```suggestion
	shell.showItemInFolder( CA_CERT_PATH );
```
Quick note that we probably want to use this method when showing the root certificate to users.